### PR TITLE
Visitor can create an article WITH CATEGORIES

### PR DESCRIPTION
**Pull request by raptorf1**
# [PT Story](https://www.pivotaltracker.com/story/show/166168028)

### Changes we propose in this PR
* Adjusts existing acceptance tests to accommodate the updated API. Adjusts `json` fixture files also.
* Adjusts WriteArticle (visitor can select out of 13 categories) & FullArticle components (displays category of article).
* Minor styling changes.
* Code formatting when needed.

### What we have learned
* `binding.pry` & `debugger` rocks!!!

### Screenshots
Screenshot from 2019-05-29 15-33-15

Screenshot from 2019-05-29 15-33-21

Screenshot from 2019-05-29 15-33-25

### CODE REVIEW

File: cypress/fixtures/list_of_articles.json [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+    "image": "https://image.freepik.com/free-photo/paris-city-view-from-high_1160-763.jpg"
+  },
+  {
+    "id": 38,
+    "title": "A Day in Rome",
+    "ingress": "How to spend 24 hours in Rome",
+    "image": "https://image.freepik.com/free-photo/roman-coliseum-seen-from-afar_1219-15.jpg"
+  },
+  {
+    "id": 9,
```
Comment (Carrosen) [inline cypress/fixtures/list_of_articles.json:21]: Nice! Just out of interest, How come you changed the id's and did not add the category_id to all of them? :)

Comment (raptorf1) [reply cypress/fixtures/list_of_articles.json:21]: These are the mock data files. Category id is not required.
We changed the id cause we were testing both with mock data and the real back-end to make sure that everything is functional. And everything is. Both with mock data and in live env.

**Review by leiter007 — COMMENTED** [review]
Looks very nice! My comment about placeholder for the drop-down has already been taken care of before I even had a chance to write it.. :)

File: src/Components/WriteArticle.jsx [hunk 2]
```diff
@@ -41,9 +50,20 @@ class WriteArticle extends Component {
...
   }
 
+  handleChangeCategory = (e, { value }) => {
+    this.setState({ category_id: value })
+  }
+
   render() {
     let message
 
+    const options = []
```
Comment (oliverochman) [inline src/Components/WriteArticle.jsx]: refactor this
```
let options = this.state.categories.map(category => {
      return {key: category.id, text: category.name, value: category.id}
    })
```

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/Components/WriteArticle.jsx (hunk 2 above)
Comment (raptorf1) [reply src/Components/WriteArticle.jsx]: done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**